### ISSUE TITLE
Support for copy to stdout

### ISSUE THREAD

**Issue by cloudharmony** [Azure/azure-storage-azcopy#288]
For performance testing it would be very helpful if copy from blob to local supported writing to stdout or at least to `/dev/null` (which is supported by `az storage blob download` but not `azcopy`). This capability would mitigate potential disk bottlenecks. For example, `aws s3` supports writing to stdout using `-` as the local path.

**Comment by zezha-msft** [comment]
Hi @cloudharmony, thanks for reaching out!

We do support stdout and `/dev/null` with the piping feature:
```bash
  - azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/blob]?[SAS]" > /dev/null
```
We'll add support for `/dev/null` in the general case soon(next release), so that you could download multiple blobs to `/dev/null` like so :
```bash
- azcopy cp "https://[account].blob.core.windows.net/[container]/?[SAS]" /dev/null --recursive
```
Please let me know if you have any other question.

**Comment by cloudharmony** [comment]
Ah, perfect, didn't realize redirection would suppress the progress and render the blob to stdout

**Comment by zezha-msft** [comment]
Hi @cloudharmony, no worries! The piping feature was designed with chained processing (such as decompression or encryption) in mind, so progress output is suppressed.

The pipe also works the other way around:
```bash
cat "/path/to/file.txt" | azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/blob]?[SAS]"
```
Please refer to `./azcopy cp --help` for more examples.

**Comment by cloudharmony** [comment]
@zezha-msft I am able to write to stdout using `azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/blob]?[SAS]" > /dev/stdout` on Ubuntu, but on CentOS I'm having trouble (writing to /dev/stdout requires sudo) - is there another method to write to stdout?

**Comment by zezha-msft** [comment]
Hi @cloudharmony, I'm not extremely familiar with CentOS, but could help to look for workarounds if you could elaborate a bit on your use case. Are you running AzCopy from a script like described [here](https://centos.org/forums/viewtopic.php?f=36&t=41076)?

**Comment by cloudharmony** [comment]
We are assessing throughput performance between Azure Virtual machines and blob storage. To do so, we use this test harness: https://github.com/cloudharmony/network. We need the `test_cmd_downlink` to support writing to stdout which is piped in to `wc -c` to verify number of bytes received (we want to avoid disk related bottlenecks). Alternatively, we can write objects to `/dev/null` and get byte count from stdout command output (e.g. via `grep`). However, with `azcopy` if we write objects to `/dev/null` via redirection, there is no way to get the command output which is suppressed by the `>` redirection. It would be great if you could support writing objects to `/dev/null` without requiring redirection and thus preserving the standard command output.

**Comment by zezha-msft** [comment]
Hi @cloudharmony, thanks for the context!

To make sure that I understood your scenario: are you executing `./azcopy cp [blob-URL] | wc -c`, and it's failing on CentOS due to permission issues? Do you mind sharing the actual command if you do something different?

**Comment by cloudharmony** [comment]
The issue I'm having is that while `azcopy cp https://[account].blob.core.windows.net/[container]/[path/to/blob] >/dev/stdout` works on Ubuntu, it does not on CentOS without `sudo`.

It would be nice if stdout content was consistent for both TTY and redirection, and local path supported `/dev/null` - for example: `azcopy cp https://[account].blob.core.windows.net/[container]/[path/to/blob] /dev/null | grep TotalBytesTransferred | grep -o -E [0-9]+` would tell me how many bytes were transferred without writing the blob to disk. I think it's a little confusing that stdout varies depending whether it is redirected (blob content) or rendered to TTY (command stats).

Thanks!

**Comment by zezha-msft** [comment]
Hi @cloudharmony, we've implemented support for `/dev/null` in 10.0.9. Thanks for the feedback!
